### PR TITLE
feat: dockerize the app

### PR DESCRIPTION
**Pull request by jackblk**
Hi,

I love your repo very much, it's a very good way to grab the freebie without any browser involved. It is much less flaky than browser way!

Also, I see that #33 has a "help wanted" tag, so here I help. This PR has no function change, only to implement a new way to use the package. Very minimal size, only 12MB bigger than `node:16-alpine3.13` image.

```
$ docker images
REPOSITORY                                   TAG             IMAGE ID       CREATED          SIZE
ghcr.io/jackblk/epicgames-freebies-claimer   1.5.0           a604af3961e7   42 minutes ago   125MB
ghcr.io/jackblk/epicgames-freebies-claimer   latest          a604af3961e7   42 minutes ago   125MB
node                                         16-alpine3.13   48de5d16038c   7 days ago       113MB
```

| image                      | **compressed (download) size** | full size |
|----------------------------|----------------------------|-----------|
| node:16-alpine3.13         | 39.15MB                    | 113MB     |
| epicgames-freebies-claimer | **41.86MB**                    | 125MB     |

I also improve some other stuff:

* change loop config to false, or container will run forever
* support latest node on latest alpine, smallest size possible
* fix format issue using eslint (default config)
* added @babel/core since this is needed for eslint-parser

I've written a [Guide on how to use the Docker image (and how to dev)](https://github.com/Revadike/epicgames-freebies-claimer/wiki/User-Guide-(Docker)).

Any feedback will be helpful 🚀

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    .gyp \
+    python3=~3.8 \
+    make=~4.3 \
+    g++=~10.2 \
+    git=~2.30 \
+    && npm install --only=production \
+    && apk del .gyp
+
+# App stage
+FROM node:16-alpine3.13 as app
```
Comment (Fzoss) [inline Dockerfile:18]: Can this build omit changes to the build environment to reduce the size of the final packaged image?

Comment (jackblk) [reply Dockerfile:18]: I'm not sure what you meant, can you propose an example or any link to the docs? I will read & optimize it.

From what I understand, currently Dockerfile is using multi-stage build, the build stage does not affect the size of the final image.

Comment (Fzoss) [reply Dockerfile:18]: I've figured it out, I didn't know about such multi-stage build dockerfile before.
I rewrote the dockerfile I made before.
The final image is much smaller, thanks

Comment (jackblk) [reply Dockerfile:18]: Great to know 😄 .
I don't know any other way to make it smaller. It's already very small to download. I will mark the conversation as resolved 🎉 .

**Comment by marchenm** [comment]
I suggest adding a `docker-compose.yaml` like:

```yaml
version: '3.4'

services:
  epicgamesfreebiesclaimer:
    image: epicgamesfreebiesclaimer
    build: 
      context: .
      dockerfile: ./Dockerfile
    volumes:
      - ./device_auths.json:/app/device_auths.json:ro
    restart: unless-stopped
```

and keeping `loop` set to `true` so you can run this in the background with `docker compose up -d`

Also optionally if @Revadike would like, add the workflow to build the image using gh-actions and use github container registry to host it in this repo. Reference that image in `docker-compose.yaml`

**Comment by Revadike** [comment]
I am not using GitHub actions. GitHub will definitely ban this repo again if they got abused, so I rather stay away from it.

**Comment by jackblk** [comment]
Thanks for the suggestion @marchenm. I'm not a fan of keeping the loop but that looks good, let's get the suggestion from @Revadike.

@Revadike 
I will remove Github Actions workflow and maintain the image on my Dockerhub & Github Package so there's no risk for you or this repo. Do you any any other concern for merging this PR?

**Comment by Revadike** [comment]
Since I'm not testing this, I have to rely on third party reviewers. If they say its ok, I will merge it.

**Comment by Revadike** [comment]
Also, could you explain why should I merge this PR and not the other docker PR?

**Comment by marchenm** [comment]
This PR uses a newer version of node, works with the latest version of the codebase, and results in a smaller image size. It also doesn't add any additional scripts to the docker image.

**Comment by Revadike** [comment]
Alright, I will likely add this with the next update. Thanks everyone contributing to and reviewing this PR!

**Pull request merged**